### PR TITLE
Add config flows for bmw_connected_drive

### PR DESCRIPTION
**Pull request by rikroe**
## Proposed change

Add documentation for config flows implemented in home-assistant/core#39585.

## Type of change

- [ ] Spelling, grammar or other readability improvements (`current` branch).
- [ ] Adjusted missing or incorrect information in the current documentation (`current` branch).
- [ ] Added documentation for a new integration I'm adding to Home Assistant (`next` branch).
  - [ ] I've opened up a PR to add logo's and icons in [Brands repository](https://github.com/home-assistant/brands).
- [x] Added documentation for a new feature I'm adding to Home Assistant (`next` branch).
- [ ] Removed stale or deprecated documentation.

## Additional information

- Link to parent pull request in the codebase: home-assistant/core#39585
- Link to parent pull request in the Brands repository: 
- This PR fixes or closes issue: 

## Checklist

- [x] This PR uses the correct branch, based on one of the following:
  - I made a change to the existing documentation and used the `current` branch.
  - I made a change that is related to an upcoming version of Home Assistant and used the `next` branch.
- [x] The documentation follows the Home Assistant documentation [standards][].

[standards]: https://developers.home-assistant.io/docs/documenting/standards

### CODE REVIEW

File: source/_integrations/bmw_connected_drive.markdown [hunk 1]
```diff
@@ -31,10 +37,16 @@ This integration provides the following platforms:
 - Services: Turn on air condition, sound the horn, flash the lights and update the state. More details can be found [here](/integrations/bmw_connected_drive/#services).
 
 ## Configuration
```
Comment (klaasnicolaas) [inline source/_integrations/bmw_connected_drive.markdown:39]:
```suggestion
## Configuration

```

File: source/_integrations/bmw_connected_drive.markdown [hunk 2]
```diff
@@ -31,10 +37,16 @@ This integration provides the following platforms:
...
+The preferred way to enable the `bmw_connected_drive`  integration is via __Configuration > Integrations__. After connecting to your account, you can set the following settings in the integration's options:
```
Comment (klaasnicolaas) [inline source/_integrations/bmw_connected_drive.markdown]:
```suggestion
The preferred way to enable the `bmw_connected_drive` integration is via **Configuration** > **Integrations**. After connecting to your account, you can set the following settings in the integration's options:
```

File: source/_integrations/bmw_connected_drive.markdown [hunk 3]
```diff
@@ -31,10 +37,16 @@ This integration provides the following platforms:
...
+
+| Setting | Description |
+|---------|-------------|
+| Read-only | No execution of services to the vehicle. Still possible to send messages and POIs via `notify` and to request a status update via `bmw_connected_drive.update_state`.
+| Use Home Assistant location for car location polls | Older cars (non i3/i8 build before 7/2014) require the phone to be close to the car to get location updates. Enable this option to use the location of your Home Assistant instance for these queries, so updates are available when your car is in the surrounding of your home. | 
 
-To enable this integration in your installation, add the following to your
-`configuration.yaml` file:
+The following settings in your `configuration.yaml` file are considered legacy. They will be imported into __Configuration > Integrations__ and you can set the options from above. Changes to `configuration.yaml` after the first import will be ignored.
```
Comment (klaasnicolaas) [inline source/_integrations/bmw_connected_drive.markdown]:
```suggestion
The following settings in your `configuration.yaml` file are considered legacy. They will be imported into **Configuration** > **Integrations** and you can set the options from above. Changes to `configuration.yaml` after the first import will be ignored. 
```

File: source/_integrations/bmw_connected_drive.markdown [hunk 4]
```diff
@@ -31,10 +37,16 @@ This integration provides the following platforms:
...
+| Setting | Description |
+|---------|-------------|
+| Read-only | No execution of services to the vehicle. Still possible to send messages and POIs via `notify` and to request a status update via `bmw_connected_drive.update_state`.
+| Use Home Assistant location for car location polls | Older cars (non i3/i8 build before 7/2014) require the phone to be close to the car to get location updates. Enable this option to use the location of your Home Assistant instance for these queries, so updates are available when your car is in the surrounding of your home. | 
 
-To enable this integration in your installation, add the following to your
-`configuration.yaml` file:
+The following settings in your `configuration.yaml` file are considered legacy. They will be imported into __Configuration > Integrations__ and you can set the options from above. Changes to `configuration.yaml` after the first import will be ignored. 
 
+### Legacy configuration
```
Comment (klaasnicolaas) [inline source/_integrations/bmw_connected_drive.markdown:50]:
```suggestion
### Legacy configuration

```

**Review by klaasnicolaas — CHANGES_REQUESTED** [review]

**Review by frenck — DISMISSED** [review]
✅ Approved. Can be merged as soon as the parent PR gets merged.

**Comment by rikroe** [comment]
Had to close the original parent PR (home-assistant/core#35732) due to completely failing at rebasing. The new parent PR is home-assistant/core#39585.

**Comment by rikroe** [comment]
Updated docs to comply with changes requested in home-assistant/core#39585.

**Pull request merged**